### ISSUE TITLE
[8193.15 DEV] Game Crashes On Module Load

### ISSUE THREAD

**Issue by pstemarie** [Beamdog/nwn-issues#156]
**To Reproduce**

 - Open the attached module in the Toolset.
 - Hit F9 to test the module
 - Game Crashes 

**Crash Dump**

[nwmain_crash.zip](https://github.com/Beamdog/nwn-issues/files/5200805/nwmain_crash.zip)

**Specifics**

 - Windows 10
 - 8193.15 Dev
 - Steam
 - Repro Rate (100%)
 - Content overrides? Module Haks

Dropbox link to module and haks as file size is too big to upload. https://www.dropbox.com/s/93o16ofvhgtyup2/ModAndHaks.zip?dl=0

**Comment by niv** [comment]
Does the crash also happen when running the module directly, not via Toolset F9?

**Comment by pstemarie** [comment]
Yes, just checked that now. Other modules load fine. So its either in my haks or in the mod file itself. The mod crashed yesterday randomly when closing the toolset. Maybe it corrupted it?

**Comment by niv** [comment]
This worked on .14 or earlier?

**Comment by pstemarie** [comment]
It was a corrupt module. I just copied the Temp0 folder from the old module to a new module and its working again. Dunno why the toolset crashed yesterday - but that was when I was still on the Preview Build. I think this issue can be closed unless you want to look into the dmp some more.

**Comment by niv** [comment]
There's definitely an issue in tno90_water01.txi: You are saying "channelscale/translate 8", but then only give 4 list elements. It then fails to parse the rest of the txi.

**Comment by pstemarie** [comment]
Hmm I had copied that from one of the vanilla TXI files. I think the TNO01_water

**Comment by niv** [comment]
Well, the worst case here probably is the water not animating properly when not using the new water shader.

There's a crash when trying to spawn the emitter vps_bugs (in area greenhaven). Wild guess based on the code, the emitter is marked as the wrong type (it expects a MdlNodeEmitter/MdlNodeAnimesh). It also can't have animations. This may be a bug in the code too - sorry, can't dig into it further right now.

**Comment by pstemarie** [comment]
No big deal, I can live without the emitter - it was a hacky way to try and add a placeable fly swarm anyway, just adding an effect to placeables.2da without actually changing the model around.

**Comment by niv** [comment]
Try without vps_bugs and see if it still crashes for you. I don't think it was ever intended for those emitters to be used that way.

**Comment by pstemarie** [comment]
Definitely a corrupt module. I took vps_bugs and pfx_bugs out of both appearance.2da and placeables.2da, removed the OnEnter script and the module still crashed. My clone module didn't crash.

**Comment by pstemarie** [comment]
OK, I've added in the recovered mod file to the zip as well as updated haks. Now I am experiencing periodic random crashes when opening an area for the first time AFTER editing a script.

To Reproduce (occurs less than 25% of the time)
1. Open the Module
2. Go down to the scripts and open one. Edit the script and save it.
3. Open an area.
4. Toolset crashes - seems to occur more often with the Greenhaven area.

Sometimes it crashes to the desktop, other times it produces a error message then crashes to the desktop before I can read the error message.

I had none of these issues in the most current Preview build.
